---
title: "ORCHIDS_LOCATION_COMPARISON"
author: "Joe Gunn"
date: "5/21/2020"
output: html_document
---
## Libraries needed for analysis
```{r libraries, echo=FALSE}
library(readxl)
library(cowplot)
library(tidyverse)
library(vegan)
library(nlme)
library(lme4)
library(factoextra)
library(viridis)
library(bbmle)
library(car)
library(multcomp)
```

## Metadata
```{r}
#Read in site-level metadata and clean
site_metadata <- read_excel("../../data/metadata/site_metadata.xlsx")

site_metadata <- site_metadata %>%
  mutate(Site = factor(Site), Near_Far = factor(Near_Far), Location = factor(Location), loc_abbr = factor(loc_abbr)) %>%
  mutate(present_absent = ifelse(as.factor(Near_Far) == "NO" | as.factor(Near_Far) == "NOE", "Absent", "Present"))

#Read in taxonomic level metadata and clean
tax_metadata <- read_excel("../../data/metadata/tax_metadata.xlsx")

tax_metadata <- tax_metadata %>%
  mutate(kingdom = factor(kingdom),
         phylum = factor(phylum), 
         class = factor(class), 
         order = factor(order), 
         family = factor(family), 
         genus = factor(genus), 
         species = factor(species))
```

## Read-in raw data
```{r}
#Read in OTU raw abundance data. This data is organized as a matrix of individual microbial OTUS (rows) and sampling sites (columns). 
soil_raw <- read_excel("../../data/excel_data/raw_data.xlsx")

#Clean table
colnames(soil_raw)[1] <- "otu_id"

soil_raw <- soil_raw %>%
  mutate(otu_id = as.character(otu_id)) %>%
  mutate(otu_id = factor(otu_id))

#Summarize raw microbial OTU data
#Get number of OTUs
otu_count <- soil_raw %>%
  nrow() #84,960 OTUs

#Get number of sites
site_count <- soil_raw %>%
  ncol() #24 sites (excluding otu id)

read_count <- soil_raw[,-c(1)] %>%
  colSums() #54,631 reads per sample - THE DATA IS ALREADY RAREFIED!


#combine taxonomy data (tax_metadatat) and raw otu data (soil_raw)
soil_all <- cbind(tax_metadata, soil_raw)
```

## Get Relative Abundance Data at the Phylum Level in a gathered dataset - this was produced in ORCHIDS_tax_confidence.rmd
```{r}
soil_data_phylum_raw <- soil_all[,-c(1,3:7)]

soil_data_phylum_rel <- soil_data_phylum_raw %>%
  mutate(LT0116S = LT0116S/54631,
         LT0216S = LT0216S/54631,
         NE0616S = NE0616S/54631,
         P0116S = P0116S/54631,
         P0216S = P0216S/54631,
         P0316S = P0316S/54631,
         H0316S = H0316S/54631,
         H0216S = H0216S/54631,
         H0116S = H0116S/54631,
         LT0516S = LT0516S/54631,
         T0416S = T0416S/54631,
         NE0216S = NE0216S/54631,
         NE0416S = NE0416S/54631,
         S0216S = S0216S/54631,
         LT0416S = LT0416S/54631,
         NE0516S = NE0516S/54631,
         S0116S = S0116S/54631,
         LT0316S = LT0316S/54631,
         NE0316S = NE0316S/54631,
         NE0116S = NE0116S/54631,
         T0316S = T0316S/54631,
         NE0816S = NE0816S/54631,
         NE0716S = NE0716S/54631,
         T0116S = T0116S/54631)

#Full Dataset with raw and relative abundance data
soil_data_phylum_top10_raw_abundance_general <- soil_data_phylum_raw %>% 
  filter(phylum == "Verrucomicrobia" | phylum == "Proteobacteria" | phylum == "Actinobacteria" | phylum == "Acidobacteria" | phylum == "Chloroflexi" | phylum == "Planctomycetes" | phylum == "Bacteroidetes" | phylum == "Firmicutes" | phylum == "Nitrospirae" | phylum == "Gemmatimonadetes")
  
soil_data_phylum_top10_relative_abundance_general <- soil_data_phylum_rel %>% 
  filter(phylum == "Verrucomicrobia" | phylum == "Proteobacteria" | phylum == "Actinobacteria" | phylum == "Acidobacteria" | phylum == "Chloroflexi" | phylum == "Planctomycetes" | phylum == "Bacteroidetes" | phylum == "Firmicutes" | phylum == "Nitrospirae" | phylum == "Gemmatimonadetes")
```

## Get raw abundance and relative abundance datasets that are ready to use for all analyses
```{r}
soil_data_phylum_top10_raw_abundance <- column_to_rownames(soil_data_phylum_top10_raw_abundance_general, "otu_id")
soil_data_phylum_top10_relative_abundance <- column_to_rownames(soil_data_phylum_top10_relative_abundance_general, "otu_id")
```

## Get separate datasets without SWALE
```{r}
#Datasets with "Far" samples removed
soil_data_phylum_top10_raw_abundance_noSWALE <- soil_data_phylum_top10_raw_abundance[,-c(15,18)]
soil_data_phylum_top10_relative_abundance_noSWALE <- soil_data_phylum_top10_relative_abundance[,-c(15,18)]
```

## Get separate datasets with only raw data and only relative abundance data, respectively
```{r}
soil_data_phylum_top10_raw_abundance_sums <- soil_data_phylum_top10_raw_abundance %>%
  group_by(phylum) %>%
  summarize_all(funs(sum))

soil_data_phylum_top10_rel_abundance_sums <- soil_data_phylum_top10_relative_abundance %>%
  group_by(phylum) %>%
  summarize_all(funs(sum))

phy_10_raw_abundance <- soil_data_phylum_top10_raw_abundance_sums %>%
  gather(LT0116S:T0116S, key = "Site", value = "raw_abundance")

phy_10_rel_abundance <- soil_data_phylum_top10_rel_abundance_sums %>%
  gather(LT0116S:T0116S, key = "Site", value = "relative_abundance")

##Clean up this dataset so that it can be joined with metadata
phy_10_raw_abundance <- phy_10_raw_abundance %>%
  left_join(site_metadata, by = "Site") #This dataset is ready to use for analyses 

##Clean up this dataset so that it can be joined with metadata
phy_10_rel_abundance <- phy_10_rel_abundance %>%
  left_join(site_metadata, by = "Site") #This dataset is ready to use for analyses 
```

## Observe differences in relative abundance of the top 10 phyla for each location to see if there are any outlier locations - SWALE is a clear outlier, with most of its relative abundance accounted for in proteobacteria rather than verrucomicrobia
## Get separate datasets for each LOCATION based on relative abundance data 

## Get separate datasets for each of the top 10 PHYLA based on raw abundance data 
```{r}
acidobacteria_raw_abundance <- phy_10_raw_abundance %>%
  filter(phylum == "Acidobacteria")

actinobacteria_raw_abundance <- phy_10_raw_abundance %>%
  filter(phylum == "Actinobacteria")

bacteroidetes_raw_abundance <- phy_10_raw_abundance %>%
  filter(phylum == "Bacteroidetes")

chloroflexi_raw_abundance <- phy_10_raw_abundance %>%
  filter(phylum == "Chloroflexi")

firmicutes_raw_abundance <- phy_10_raw_abundance %>%
  filter(phylum == "Firmicutes")

gemmatimonadetes_raw_abundance <- phy_10_raw_abundance %>%
  filter(phylum == "Gemmatimonadetes")

nitrospirae_raw_abundance <- phy_10_raw_abundance %>%
  filter(phylum == "Nitrospirae")

planctomycetes_raw_abundance <- phy_10_raw_abundance %>%
  filter(phylum == "Planctomycetes")

proteobacteria_raw_abundance <- phy_10_raw_abundance %>%
  filter(phylum == "Proteobacteria")

verrucomicrobia_raw_abundance <- phy_10_raw_abundance %>%
  filter(phylum == "Verrucomicrobia")
```

## Get mean relative abundance for each phylum
```{r}
#Get mean relative abundance for each of the top 10 phyla
phy_10_rel_abundance_phy_means <- phy_10_rel_abundance %>%
  group_by(phylum) %>%
  summarize(mean_rel = mean(relative_abundance))

#Bind the mean of the top 10 phyla to the abundance dataset so that they can be mapped onto the same figure
phy_10_rel_abundance <- phy_10_rel_abundance %>%
  left_join(phy_10_rel_abundance_phy_means, by = "phylum")
```


```{r}
phy_10_rel_abundance$phylum <- factor(phy_10_rel_abundance$phylum, levels = c("Verrucomicrobia","Proteobacteria","Actinobacteria","Acidobacteria","Chloroflexi","Planctomycetes", "Bacteroidetes","Firmicutes","Nitrospirae","Gemmatimonadetes"))

pdf("/Users/joegunn/Desktop/Grad_School_Stuff/Research/Projects/Soil_Microbiome/visualization/phylum_comparison_figures/location_comparisons_by_phylum.pdf", width = 10, height = 12)

ggplot(phy_10_rel_abundance, aes(x = loc_abbr, y = relative_abundance, fill = Location)) + 
  geom_boxplot(show.legend = F, outlier.shape = NA, alpha = 0.6) + 
  geom_point(position = position_jitter(width = 0.2), alpha = 0.6, color = "black", pch = 21, show.legend = F) +
  geom_hline(aes(yintercept = mean_rel), linetype = "longdash", size = 0.5) +
  theme_set(theme_cowplot(12)) +
  facet_wrap(~phylum, ncol = 2) +
  scale_fill_brewer(palette = "Dark2") +
  labs(x = "Location", y = "Relative Abundance") +
  theme(strip.background = element_rect(fill = "white")) +
  theme(strip.text = element_text(size = 15)) +
  theme(panel.background = element_rect(color = "black", fill = "white", size = 1, linetype = "solid")) +
  theme(panel.margin = unit(1, "lines")) +
  theme(axis.title = element_text(size = 15)) +
  theme(axis.text = element_text(size = 12)) +
  theme(axis.line = element_blank()) +
  theme(axis.ticks.x = element_blank())  +
  theme(axis.title.x = element_blank())

dev.off()
```

```{r}
## Visualize and run general linear mixed effect models (with location as a random effect in each model) for each phylum to assess differences among different orchid LOCATION treatments
acidobacteria_glmer <- glmer.nb(raw_abundance ~ loc_abbr + (1|Near_Far), data = acidobacteria_raw_abundance)
actinobacteria_glmer <- glmer.nb(raw_abundance ~ loc_abbr + (1|Near_Far), data = actinobacteria_raw_abundance)
bacteroidetes_glmer <- glmer.nb(raw_abundance ~ loc_abbr + (1|Near_Far), data = bacteroidetes_raw_abundance)
chloroflexi_glmer <- glmer.nb(raw_abundance ~ loc_abbr + (1|Near_Far), data = chloroflexi_raw_abundance)
firmicutes_glmer <- glmer.nb(raw_abundance ~ loc_abbr + (1|Near_Far), data = firmicutes_raw_abundance)
gemmatimonadetes_glmer <- glmer.nb(raw_abundance ~ loc_abbr + (1|Near_Far), data = gemmatimonadetes_raw_abundance)
nitrospirae_glmer <- glmer.nb(raw_abundance ~ loc_abbr + (1|Near_Far), data = nitrospirae_raw_abundance)
planctomycetes_glmer <- glmer.nb(raw_abundance ~ loc_abbr + (1|Near_Far), data = planctomycetes_raw_abundance)
proteobacteria_glmer <- glmer.nb(raw_abundance ~ loc_abbr + (1|Near_Far), data = proteobacteria_raw_abundance)
verruco_glmer <- glmer.nb(raw_abundance ~ loc_abbr + (1|Near_Far), data = verrucomicrobia_raw_abundance)
```

## All model summary outputs - with p values corrected for false discovery rate 
```{r}
#Get p-values from each Anova output object - these are the overall p-values for the whole model, not separated by pairwise comparisons
acido_p <- Anova(acidobacteria_glmer)$"Pr(>Chisq)" #SIGNIFICANT
actino_p <- Anova(actinobacteria_glmer)$"Pr(>Chisq)"
bact_p <- Anova(bacteroidetes_glmer)$"Pr(>Chisq)" #SIGNIFICANT
chlor_p <- Anova(chloroflexi_glmer)$"Pr(>Chisq)" #SIGNIFICANT
firm_p <- Anova(firmicutes_glmer)$"Pr(>Chisq)" #SIGNIFICANT
gemma_p <- Anova(gemmatimonadetes_glmer)$"Pr(>Chisq)" #SIGNIFICANT
nitro_p <- Anova(nitrospirae_glmer)$"Pr(>Chisq)" #SIGNIFICANT
plancto_p <- Anova(planctomycetes_glmer)$"Pr(>Chisq)" #SIGNIFICANT
proteo_p <- Anova(proteobacteria_glmer)$"Pr(>Chisq)" #SIGNIFICANT
verruco_p <- Anova(verruco_glmer)$"Pr(>Chisq)" #SIGNIFICANT

#Gather all p-values from the global models into a single dataframe 
loc_glm_p_vals <- data.frame(p_val = c(acido_p, actino_p, bact_p, chlor_p, firm_p, gemma_p, nitro_p, plancto_p, proteo_p, verruco_p))

#Run false discovery rate multiple comparison adjustment on p-values from global models - all values come out significant still, with the exception of Actinobacteria, which was not signficant to begin with.
loc_glm_p_vals_fdr <- p.adjust(loc_glm_p_vals$p_val, method = "fdr")
```

## All model post-hoc analyses
```{r}
post_acidobacteria <- glht(acidobacteria_glmer, mcp(loc_abbr = "Tukey"))
summary(post_acidobacteria) 

post_actinobacteria <- glht(actinobacteria_glmer, mcp(loc_abbr = "Tukey"))
summary(post_actinobacteria)

post_bacteroidetes <- glht(bacteroidetes_glmer, mcp(loc_abbr = "Tukey"))
summary(post_bacteroidetes) 

post_chloroflexi <- glht(chloroflexi_glmer, mcp(loc_abbr = "Tukey"))
summary(post_chloroflexi) 

post_firmicutes <- glht(firmicutes_glmer, mcp(loc_abbr = "Tukey"))
summary(post_firmicutes) 

post_gemmatimonadetes <- glht(gemmatimonadetes_glmer, mcp(loc_abbr = "Tukey"))
summary(post_gemmatimonadetes) 

post_nitrospirae <- glht(nitrospirae_glmer, mcp(loc_abbr = "Tukey"))
summary(post_nitrospirae) 

post_planctomycetes <- glht(planctomycetes_glmer, mcp(loc_abbr = "Tukey"))
summary(post_planctomycetes) 

post_proteobacteria <- glht(proteobacteria_glmer, mcp(loc_abbr = "Tukey"))
summary(post_proteobacteria) 

post_verruco <- glht(verruco_glmer, mcp(loc_abbr = "Tukey"))
summary(post_verruco) 
```