---
title: "ORCHIDS_DISTANCE_COMPARISON"
author: "Joe Gunn"
date: "5/21/2020"
output: html_document
---

## Libraries needed for analysis
```{r libraries, echo=FALSE}
library(readxl)
library(cowplot)
library(tidyverse)
library(vegan)
library(nlme)
library(lme4)
library(factoextra)
library(viridis)
library(bbmle)
library(car)
library(multcomp)
library(plyr)
```


## Metadata
```{r}
#Read in site-level metadata and clean
site_metadata <- read_excel("../../data/metadata/site_metadata.xlsx")

site_metadata <- site_metadata %>%
  mutate(Site = factor(Site), Near_Far = factor(Near_Far), Location = factor(Location), loc_abbr = factor(loc_abbr)) %>%
  mutate(present_absent = ifelse(as.factor(Near_Far) == "NO" | as.factor(Near_Far) == "NOE", "Absent", "Present"))

#Read in taxonomic level metadata and clean
tax_metadata <- read_excel("../../data/metadata/tax_metadata.xlsx")

tax_metadata <- tax_metadata %>%
  mutate(kingdom = factor(kingdom),
         phylum = factor(phylum), 
         class = factor(class), 
         order = factor(order), 
         family = factor(family), 
         genus = factor(genus), 
         species = factor(species))
```

## Read-in raw data
```{r}
#Read in OTU raw abundance data. This data is organized as a matrix of individual microbial OTUS (rows) and sampling sites (columns). 
soil_raw <- read_excel("../../data/excel_data/raw_data.xlsx")

#Clean table
colnames(soil_raw)[1] <- "otu_id"

soil_raw <- soil_raw %>%
  mutate(otu_id = as.character(otu_id)) %>%
  mutate(otu_id = factor(otu_id))

#Summarize raw microbial OTU data
#Get number of OTUs
otu_count <- soil_raw %>%
  nrow() #84,960 OTUs

#Get number of sites
site_count <- soil_raw %>%
  ncol() #24 sites (excluding otu id)

read_count <- soil_raw[,-c(1)] %>%
  colSums() #54,631 reads per sample - THE DATA IS ALREADY RAREFIED!


#combine taxonomy data (tax_metadatat) and raw otu data (soil_raw)
soil_all <- cbind(tax_metadata, soil_raw)
```

## Get Relative Abundance Data at the Phylum Level in a gathered dataset - this was produced in ORCHIDS_tax_confidence.rmd
```{r}
soil_data_phylum_raw <- soil_all[,-c(1,3:7)]

soil_data_phylum_rel <- soil_data_phylum_raw %>%
  mutate(LT0116S = LT0116S/54631,
         LT0216S = LT0216S/54631,
         NE0616S = NE0616S/54631,
         P0116S = P0116S/54631,
         P0216S = P0216S/54631,
         P0316S = P0316S/54631,
         H0316S = H0316S/54631,
         H0216S = H0216S/54631,
         H0116S = H0116S/54631,
         LT0516S = LT0516S/54631,
         T0416S = T0416S/54631,
         NE0216S = NE0216S/54631,
         NE0416S = NE0416S/54631,
         S0216S = S0216S/54631,
         LT0416S = LT0416S/54631,
         NE0516S = NE0516S/54631,
         S0116S = S0116S/54631,
         LT0316S = LT0316S/54631,
         NE0316S = NE0316S/54631,
         NE0116S = NE0116S/54631,
         T0316S = T0316S/54631,
         NE0816S = NE0816S/54631,
         NE0716S = NE0716S/54631,
         T0116S = T0116S/54631)

#Full Dataset with raw and relative abundance data
soil_data_phylum_top10_raw_abundance_general <- soil_data_phylum_raw %>% 
  filter(phylum == "Verrucomicrobia" | phylum == "Proteobacteria" | phylum == "Actinobacteria" | phylum == "Acidobacteria" | phylum == "Chloroflexi" | phylum == "Planctomycetes" | phylum == "Bacteroidetes" | phylum == "Firmicutes" | phylum == "Nitrospirae" | phylum == "Gemmatimonadetes")
  
soil_data_phylum_top10_relative_abundance_general <- soil_data_phylum_rel %>% 
  filter(phylum == "Verrucomicrobia" | phylum == "Proteobacteria" | phylum == "Actinobacteria" | phylum == "Acidobacteria" | phylum == "Chloroflexi" | phylum == "Planctomycetes" | phylum == "Bacteroidetes" | phylum == "Firmicutes" | phylum == "Nitrospirae" | phylum == "Gemmatimonadetes")
```

## Get raw abundance and relative abundance datasets that are ready to use for all analyses
```{r}
soil_data_phylum_top10_raw_abundance <- column_to_rownames(soil_data_phylum_top10_raw_abundance_general, "otu_id")
soil_data_phylum_top10_relative_abundance <- column_to_rownames(soil_data_phylum_top10_relative_abundance_general, "otu_id")
```

## Get separate datasets without SWALE
```{r}
#Datasets with "Far" samples removed
soil_data_phylum_top10_raw_abundance_noSWALE <- soil_data_phylum_top10_raw_abundance[,-c(15,18)]
soil_data_phylum_top10_relative_abundance_noSWALE <- soil_data_phylum_top10_relative_abundance[,-c(15,18)]
```

## From this point on, Site "SWALE" was excluded from analyses due to being a significant outlier in composition of the top 10 phyla and for low sample size (only two samples within the site)
## Get separate datasets with only raw data and only relative abundance data, respectively
```{r}
soil_data_phylum_top10_raw_abundance_sums <- soil_data_phylum_top10_raw_abundance_noSWALE %>%
  group_by(phylum) %>%
  summarize_all(funs(sum))

soil_data_phylum_top10_rel_abundance_sums <- soil_data_phylum_top10_relative_abundance_noSWALE %>%
  group_by(phylum) %>%
  summarize_all(funs(sum))

phy_10_raw_abundance <- soil_data_phylum_top10_raw_abundance_sums %>%
  gather(LT0116S:T0116S, key = "Site", value = "raw_abundance")

phy_10_rel_abundance <- soil_data_phylum_top10_rel_abundance_sums %>%
  gather(LT0116S:T0116S, key = "Site", value = "relative_abundance")

##Clean up this dataset so that it can be joined with metadata
phy_10_raw_abundance <- phy_10_raw_abundance %>%
  left_join(site_metadata, by = "Site") #This dataset is ready to use for analyses 

##Clean up this dataset so that it can be joined with metadata
phy_10_rel_abundance <- phy_10_rel_abundance %>%
  left_join(site_metadata, by = "Site") #This dataset is ready to use for analyses 
```

```{r}
###All datasets below are for individual phyla (top 10) and show raw abundance data
phy_10_raw_abundance_withOrchids <- phy_10_raw_abundance %>%
  filter(!(Near_Far == "NO")) %>% 
  filter(!(Near_Far == "NOE")) %>%
  mutate(log_abundance = log10(raw_abundance))

levels(phy_10_raw_abundance_withOrchids$Near_Far)[levels(phy_10_raw_abundance_withOrchids$Near_Far) == "Near"] <- "0"
levels(phy_10_raw_abundance_withOrchids$Near_Far)[levels(phy_10_raw_abundance_withOrchids$Near_Far) == "Intermediate"] <- "100"
levels(phy_10_raw_abundance_withOrchids$Near_Far)[levels(phy_10_raw_abundance_withOrchids$Near_Far) == "Far"] <- "200"

phy_10_raw_abundance_withOrchids <- phy_10_raw_abundance_withOrchids %>%
  mutate(Near_Far = as.character(Near_Far)) %>%
  mutate(Near_Far = as.double(Near_Far))

#Plot overall effect of distance on phylum abundance, coloring each set of points by their corresponding phylum
pdf("/Users/joegunn/Desktop/Grad_School_Stuff/Research/Projects/Soil_Microbiome/visualization/distance_comparison_figures/all_otus_distance_plot.pdf", width = 8, height = 5)

ggplot(phy_10_raw_abundance_withOrchids, aes(x = Near_Far, y = log_abundance, fill = phylum)) +
  geom_point(size = 3, alpha = 0.9, position = position_jitter(width = 5), pch = 21, color = "black", show.legend = F) + 
  geom_smooth(aes(color = phylum), method = lm, se = F, show.legend = F) +
  labs(x = "Distance from Orchid (cm)", y = "Log Abundance", fill = "Phylum", color = "Phylum") +
  scale_fill_viridis(discrete = T, option = "viridis") + 
  scale_color_viridis(discrete = T, option = "viridis") +
  theme_set(theme_cowplot(12)) +
  theme(legend.background = element_rect(color = "black")) +
  theme(legend.margin = margin(3,3,3,3)) +
  theme(legend.title = element_text(size = 15, face = "bold")) +
  theme(legend.text = element_text(size = 15)) +
  theme(axis.title = element_text(size = 15)) +
  theme(axis.text = element_text(size = 15))
  
dev.off()
```

## Run general linear mixed effect models with a negative bionomial distribution on raw abundance data
```{r}
distance_glm_phy_random <- glmer.nb(raw_abundance ~ Near_Far + (1|phylum), data =  phy_10_raw_abundance_withOrchids)
Anova(distance_glm_phy_random) #Very insignificant p = 0.851

distance_glm_location_random <- glmer.nb(raw_abundance ~ Near_Far + (1|Location), data =  phy_10_raw_abundance_withOrchids)
Anova(distance_glm_location_random) #Very insignificant p = 0.999
```
